### PR TITLE
Web3.eth.subscribe syncing test

### LINKED ISSUE #5242: Web3.eth.subscribe syncing test
Before start check [this PR](https://github.com/ChainSafe/web3.js/pull/5005)

System tests for:

## Web3.eth.subscribe

- [x] subscribe(“syncing”)

#### Min Platform Requirements for 4.x Alpha:
- Node: v14, v16
- Browsers: Firefox, Brave
- Eth Clients: Geth, Ganache

More details are in [Epic](https://github.com/ChainSafe/web3.js/issues/4711)

related to [#4990](https://app.zenhub.com/workspaces/web3js-553a980ab285bae538012a40/issues/chainsafe/web3.js/4990)

### PR DESCRIPTION
**Pull request by avkos**
## Description

Fixes #5242 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -128,6 +128,28 @@ jobs:
       -   run: yarn test:e2e:${{ matrix.backend }}:${{ matrix.mode }}
           shell: bash
 
+  e2e-sync:
+      name: Integration Sync Test
+      needs: build
+      runs-on: ubuntu-latest
+      env:
+          INFURA_GOERLI_HTTP: ${{ secrets.INFURA_GOERLI_HTTP }}
+          INFURA_GOERLI_WS: ${{ secrets.INFURA_GOERLI_WS }}
```
Comment (jdevcs) [inline .github/workflows/build.yml]: these are not used in e2e-sync. so should be removed.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -128,6 +128,28 @@ jobs:
...
+      strategy:
+          fail-fast: false
+          matrix:
+              node: [ 14, 16 ]
```
Comment (jdevcs) [inline .github/workflows/build.yml]: I think we can reduce some CI time, by running this test on node 16 only.

File: .github/workflows/build.yml (hunk 1 above)
Comment (avkos) [reply .github/workflows/build.yml]: done

File: .github/workflows/build.yml (hunk 2 above)
Comment (avkos) [reply .github/workflows/build.yml]: done

**Review by jdevcs — APPROVED** [review]

**Review by Spacesai1or — APPROVED** [review]

**Pull request merged**